### DISCUSSION TITLE
how do i put multiple commands in entrypoint along with /sbin/init?

### DISCUSSION THREAD

**Discussion by imperialguy** [containers/podman#13041] [Q&A]
The following will result in a container that exits immediately. 

```
podman run -d  --entrypoint "/bin/sh" alpine -c '/sbin/init && echo hello'
```

I don't want the container exiting. For example, if I do the following, the container stays up and running. 

```
podman run -d --entrypoint "/bin/sh" alpine -c '/sbin/init'
```

But, what I want is to be able to run commands after the `/sbin/init`.

Had to re-create this issue because #13026 was closed

**Comment by vrothberg** [comment]
@imperialguy, you can run `podman exec` on the container. Is that an alternative for what you want to do?

**Comment by imperialguy** [comment]
`podman exec` is an external stimulus. I am looking for something that will happen at the time of container initialization, without me having to trigger it externally. I even tried this, for e.g.:

```
podman run -d --name testc --entrypoint "/sbin/init" alpine:latest -c 'echo hellow > /root/sample'
```

The above will spin up a container that doesn't exit immediately. But, it doesn't create the `/root/sample` file during container init which is what I expect. So, this what happens:

```
podman exec testc ls -l /root/sample
ls: /root/sample: No such file or directory
```

**Comment by vrothberg** [comment]
I don't think that'll work since `/sbin/init` cannot run any command.

@rhatdan @edsantiago any ideas?

**Comment by edsantiago** [comment]
I've always just created a script with a sequence of commands, copied the script into the image, and used the script as the entrypoint.

**Comment by imperialguy** [comment]
I tried that as well. It exits the container. Here:

```
podman run -d -v /home/testuser/scratch:/root/scratch --name testc --entrypoint "/root/scratch/entry.sh" alpine:latest
```

Where, `entry.sh` just has the following two lines:

```
/sbin/init
echo "hello world" > /root/sample
```

But, here's the result. Basically, it exits the container immediately. 

```
podman ps -a

CONTAINER ID  IMAGE                            COMMAND     CREATED         STATUS                     PORTS       NAMES
8226cebdd5e8  docker.io/library/alpine:latest              15 seconds ago  Exited (1) 16 seconds ago              testc
```

Is there something I am missing inside my `entry.sh`?

**Comment by vrothberg** [comment]
I am not sure that's going to work if we want to have `/sbin/init` as PID 1.  If the script is run by `sh`, it'll be PID 1.

**Comment by rhatdan** [comment]
This is a discussion not an issue.
